website/content/en/docs/reference/configuration/unit-tests.md

```diff
@@ -0,0 +1,359 @@
+`inputs` and `outputs`.
+
+{{< success title="Multiple config formats available" >}}
+The unit testing example above is in TOML but Vector also supports YAML and JSON as configuration
+formats.
+{{< /success >}}
+
+## Configuring unit tests {#configuring}
+
+Unit tests in Vector live alongside your topology configuration. You can specify your tests in the
+same config file alongside your transform definitions or split them out into a separate file.
+
+Unit tests need are specified inside of a `tests` array. Each test requires a `name`:
+
+```toml
+[[tests]]
+name = "test 1"
+# Other test config
+
+[[tests]]
+name = "test_2"
+# Other test config
+
+# etc.
+```
+
+Inside each test definition, you need to specify two things:
+
+* An array of `inputs` that provides [input events](#inputs) for the test.
+* An array of `outputs` that provides [expected outputs](#outputs) for the test.
+
+Optionally, you can specify a `no_outputs_from` list of transforms that must *not* output events
+in order for the test to pass. Here's an example:
+
+```toml
+[[tests]]
+name = "skip_modify_transform"
+no_outputs_from = ["modify_transform"]
+```
+
+In this case,
+
+### Inputs
+
+In in the `inputs` array for the test, you have these options:
+
+Parameter | Type | Description
+:---------|:-----|:-----------
+`insert_at` | string (name of transform) | The name of the transform into which the test input is inserted. This is particularly useful when you want to test only a subset of a transform pipeline.
+`value` | string (raw event value) | A raw string value to act as an input event. Use only in cases where events are raw strings and not structured objects with event fields.
+`log_fields` | object | If the transform handles [log events](#logs), these are the key/value pairs that comprise the input event.
+`metric` | object | If the transform handles [metric events](#metrics), these are the fields that comprise that metric. Subfields include `name`, `tags`, `kind`, and others.
+
+Here's an example `inputs` declaration:
+
+```toml
+[transforms.add_metadata]
+# transform config
+
+[[tests]]
+name = "Test add_metadata transform"
+
+[[tests.inputs]]
+insert_at = "add_metadata"
+
+[tests.inputs.log_fields]
+message = "<102>1 2020-12-22T15:22:31.111Z vector-user.biz su 2666 ID389 - Something went wrong"
+tags.environment = "production"
+```
+
+### Outputs
+
+In the `outputs` array of your unit testing configuration you specify two things:
+
+Parameter | Type | Description
+:---------|:-----|:-----------
+`extract_from` | string (name of transform) | The transform whose output you want to test.
+`conditions` | array of objects | The [VRL conditions][verifying] to run against the output.
+
+Each condition in the `conditions` array has two fields:
+
+Parameter | Type | Description
+:---------|:-----|:-----------
+`type` | string | The type of condition you're providing. As the original `check_fields` syntax is now deprecated, this defaults to  `vrl`.
```

Comment:
Technically I think `datadog_search` is supported here too because it is a "condition", but I doubt that would be useful for unit tests.

Reply:
I'm thinking maybe we can wait on some indication of that would fulfill a real user need before we document that